server/sonar-server/src/main/java/org/sonar/server/user/ws/UserJsonWriter.java

```diff
@@ -94,7 +93,7 @@ public void write(JsonWriter json, @Nullable UserDto user) {
   }
 
   private void writeGroupsIfNeeded(JsonWriter json, Collection<String> groups, @Nullable Collection<String> fields) {
-    if (isFieldNeeded(FIELD_GROUPS, fields) && userSession.hasPermission(GlobalPermissions.SYSTEM_ADMIN)) {
+    if (isFieldNeeded(FIELD_GROUPS, fields) && userSession.isRoot()) {
```

Comment:
I know it's pretty much the existing behavior, but it seems to me the currently authenticated user can't see its own groups with this approach unless she's an admin... and now unless she's root.

I find this unexpected